plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1106,6 +1106,31 @@ public void testRollbackSnapshot()
         dropTable("test_rollback");
     }
 
+    @Test
+    public void testRollbackSnapshotWithSchemaEvolution()
+    {
+        assertUpdate("CREATE TABLE test_rollback_with_schema_evolution AS SELECT 1 col0", 1);
+        long afterCreateTableId = getCurrentSnapshotId("test_rollback_with_schema_evolution");
+
+        assertUpdate("ALTER TABLE test_rollback_with_schema_evolution ADD COLUMN col1 int");
+        assertUpdate("INSERT INTO test_rollback_with_schema_evolution VALUES (2, 2)", 1);
+
+        assertQuery("SELECT * FROM test_rollback_with_schema_evolution", "VALUES (1, NULL), (2, 2)");
+
+        assertQuery("SELECT * FROM test_rollback_with_schema_evolution FOR VERSION AS OF " + afterCreateTableId, "VALUES 1");
+
+        assertUpdate("CALL system.rollback_to_snapshot('tpch', 'test_rollback_with_schema_evolution', " + afterCreateTableId + ")");
+
+        // Verify the output doesn't have col1 column after rollback
+        assertThat(query("DESCRIBE test_rollback_with_schema_evolution"))
+                .projected(0)
+                .skippingTypesCheck()
+                .matches("VALUES 'col0'");
+        assertQuery("SELECT * FROM test_rollback_with_schema_evolution", "VALUES 1");
+
```

Comment:
add

```java
// A DML should not change the table schema (which could be possible if the rolled back schema gets revived when committing th DML)
assertUpdate("INSERT INTO test_rollback_with_schema_evolution VALUES (13)");
assertThat(query("DESCRIBE test_rollback_with_schema_evolution"))
        .projected(0)
        .skippingTypesCheck()
        .matches("VALUES 'col0'");
assertQuery("SELECT * FROM test_rollback_with_schema_evolution", "VALUES 1, 13");
```

This should pass, but actually the INSERT is failing.

It seems to me now that we cannot fix https://github.com/trinodb/trino/issues/13699 without Iceberg side being fixed (https://github.com/apache/iceberg/issues/5591).